src/server/auth/server/api_server.go

```diff
@@ -83,8 +83,13 @@ type apiServer struct {
 	watchesEnabled bool
 }
 
+type InternalAuthServer struct {
+	APIServer    authiface.APIServer
```

Comment:
I think you might want to do `type InternalAuthServer struct { authiface.APIServer; EnvBootstrap ... }` so that's it's explicit we're trying to wrap an authiface.APIServer here.  You can then use `InternalAuthServer` like it's any `authiface.APIServer`, and avoid most of the refactoring below.